src/main/java/com/github/manolo8/darkbot/modules/DisconnectModule.java

```diff
@@ -22,15 +28,30 @@ public class DisconnectModule extends TemporalModule {
 
     private Long pauseUntil = null;
     private boolean refreshing = false;
+    private boolean closeBot = false;
 
     /**
-     * @param pauseTime null for infinite pause, otherwise pause for that amount of MS.
+     * @param pauseTime null for infinite pause, otherwise pause for that amount of
+     *                  MS.
      */
     public DisconnectModule(Long pauseTime, String reason) {
         this.reason = reason;
         this.pauseTime = pauseTime;
     }
 
+    /**
+     * @param pauseTime null for infinite pause, otherwise pause for that amount of
+     *                  MS.
+     */
+    public DisconnectModule(Long pauseTime, String reason, boolean closeBot) {
+        this(pauseTime, reason);
+        this.closeBot = closeBot;
+
+        if (closeBot) {
+            showClosePopup();
+        }
```

Comment:
This is showing the pop-up when the module is created, not when the module is set/activated to run.
It should be part of the tick function to run this

Reply:
I've put it here so that the shutdown timer itself will time the shutdown to prevent it from closing.

Reply:
I don't think you understand the issue:

```java
class MyFeature implements Behavior {

    private final DisconnectModule shutdown = new DisconnectModule(0L, "Shutting down bot", true);

    public void tick() {
        if (someConditionMet) bot.setModule(shutdown);
    }
}

```

This is a valid to create/initialize/setup the disconnect module, however this would show the pop-up instantly on module creation, instead of when "someConditionMet" which is when it actually starts attempting the disconnect.

Even if that itself is fixed, it would still shutdown in 0s if it was already near the portal giving no time to the user to react

Reply:
I didn't take that into account, I have just fixed it.